crypto/lhash/lh_stats.c

```diff
@@ -10,19 +10,16 @@
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
-/*
- * If you wish to build this outside of OpenSSL, remove the following lines
- * and things should work as expected
- */
 #include "internal/cryptlib.h"
 
 #include <openssl/bio.h>
 #include <openssl/lhash.h>
 #include "lhash_local.h"
 
-# ifndef OPENSSL_NO_STDIO
+#ifndef OPENSSL_NO_STDIO
 void OPENSSL_LH_stats(const OPENSSL_LHASH *lh, FILE *fp)
 {
+# ifndef OPENSSL_NO_DEPRECATED_3_0
```

Comment:
It looks wrong to drop just the code of the functions and not the complete functions. If some functionality is deprecated, the related APIs should be dropped completely with no-deprecated build.

Reply:
There was a comment in lhash.h:
```
 * We would like to deprecate these functions, but doing so gets errors in
 * builds when allowing deprecated, and we'd have to annotate this file
 * which is include'd in several other files, so that's not good.
 * Instead, we just declare these as normal, and the |DEFINE_LASH_OF|
 * macro includes the type-safe wrapper functions or not, depending on
 * deprecation.
```

But I just tried ```config``` and ```config no-deprecated``` and both builds worked so I guess along the way I fixed something.

Fixup commit pushed.